src/controllers/authController.js

```diff
@@ -13,31 +14,47 @@ const register = async (req, res) => {
     throw new CustomError.BadRequestError('Please provide all required fields');
   }
 
-  const emailAlreadyExists = await User.findOne({ email }); // Check if a user with the email already exists
-  if (emailAlreadyExists) {
-    throw new CustomError.BadRequestError('Email already exists'); // If user exists, throw an error
-  }
-
-  const isFirstAccount = (await User.countDocuments({})) === 0; // Check if it's the first account
-  const role = isFirstAccount ? 'admin' : 'user'; // Assign a role based on first account or not
-
-  const user = await User.create({
-    name,
-    username,
-    email,
-    password,
-    role,
-  }); // Create a new user in the database
+  const session = await mongoose.startSession();
+  session.startTransaction();
 
-  // Send the welcome email
-  await emailSender.sendWelcomeEmail(user.email, user);
+  try {
+    const emailAlreadyExists = await User.findOne({ email }); // Check if a user with the email already exists
```

Comment:
`session.startTransaction` should be inside our try block so that errors that happen within that transaction will be properly caught by the `try/catch` block: https://www.ultimateakash.com/blog-details/IiwzQGAKYAo=/How-to-implement-Transactions-in-Mongoose-&-Node.Js-(Express)

```js
  const session = await mongoose.startSession();

  try {
    session.startTransaction();
    const emailAlreadyExists = await User.findOne({ email }); // Check if a user with the email already exists
```